src/java/org/apache/cassandra/utils/EstimatedHistogram.java

```diff
@@ -365,6 +370,57 @@ public int hashCode()
         return Objects.hashCode(getBucketOffsets(), getBuckets(false));
     }
 
+    public int size()
+    {
+        return toIntExact(count());
+    }
+
+    public void update(long l)
+    {
+        this.add(l);
+    }
+
+    static class EstimatedHistogramSnapshot extends UniformSnapshot
```

Comment:
this class doesn't actually snapshot the data (mutations to the backing histogram after snapshot impact the snapshot's results), and produces confusing results (longs mean different things).  

EstimatedHistogram buckets represent their values and the long in them are the counts of those values, UniformSnapshot expects the longs to be the values themselves.

A better way to deal with this is to keep this class, but snapshot by calling `new EstimatedHistogram(histogram.getBuckets(false))` and have this class extend `com.codahale.metrics.Snapshot`  directly.

Reply:
this was copied from previous implementation that was replaced with the decaying reservoir. I can fix this up to be correct